specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json

```diff
@@ -749,34 +827,111 @@
         }
       ]
     },
+    "DiskPoolCreate": {
+      "type": "object",
+      "description": "Request payload for create or update Disk Pool request.",
+      "required": [
+        "properties",
+        "location"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties for Disk Pool create request.",
+          "$ref": "#/definitions/DiskPoolCreateProperties"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ],
+          "description": "Resource tags."
+        },
+        "location": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "The geo-location where the resource lives."
+        },
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Fully qualified resource Id for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the resource"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of the resource. Ex- Microsoft.Compute/virtualMachines or Microsoft.Storage/storageAccounts."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "DiskPoolUpdate": {
+      "type": "object",
+      "description": "Request payload for Update Disk Pool request.",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties for Disk Pool update request.",
+          "$ref": "#/definitions/DiskPoolUpdateProperties"
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ],
+          "description": "Resource tags."
+        }
+      }
+    },
     "AvailabilityZone": {
       "type": "string",
-      "description": "Availability zone"
+      "description": "Availability zone."
     },
     "DiskPoolProperties": {
       "type": "object",
-      "description": "Properties section of the Disk Pool request payload.",
+      "description": "Disk Pool response properties.",
       "required": [
         "availabilityZones",
-        "subnetId"
+        "subnetId",
+        "provisioningState",
```

Comment:
I think provisioningState and status are set by the service and are not set by a user, consider 2 things: 1. Marking properties set by the service as readOnly: true and 2. Removing them from "required"

Reply:
The customer should be using DiskPoolCreate or DiskPoolUpdate to write to the resource and not DiskPool. We did this refactoring in response to feedback from one of the tooling teams.

The RP uses the DiskPool object as part of the response payloads for GET, PUT, PATCH requests on a Disk pool. I've marked the provisioningState and status as required -- to communicate that the RP is responsible for populating all these fields as part of the response payload.